src/lib/scheduler/topology.hpp

```diff
@@ -29,15 +29,12 @@ struct TopologyNode final {
  */
 class Topology final {
  public:
-  /**
-   * @param max_num_workers A values of zero indicates no limit
-   * @param workers_per_node
-   */
-  static std::shared_ptr<Topology> create_fake_numa_topology(uint32_t max_num_workers = 0,
-                                                             uint32_t workers_per_node = 1);
-  static std::shared_ptr<Topology> create_numa_topology(uint32_t max_num_cores = 0);
+  static Topology& current();
 
-  Topology(std::vector<TopologyNode>&& nodes, size_t num_cpus) : _nodes(std::move(nodes)), _num_cpus(num_cpus) {}
+  void create_default_topology();
+  void create_numa_topology(uint32_t max_num_cores = 0);
+  void create_nonnuma_topology(uint32_t max_num_cores = 0);
```

Comment:
Really minor, but I'd prefer `*_non_numa_*` since there is no such word "nonnuma"